cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillRunnerService.java

```diff
@@ -115,17 +115,21 @@ public KubeTwillRunnerService(String kubeNamespace, DiscoveryServiceClient disco
     // Selects all runs start by the k8s twill runner that has the run id label
     String selector = String.format("%s=%s,%s", RUNNER_LABEL, RUNNER_LABEL_VAL, RUN_ID_LABEL);
     this.kubeResourceWatcher = KubeResourceWatcher.createDeploymentWatcher(kubeNamespace, selector);
+    this.objectMetaProvider = getObjectMetaProvider(kubeResourceType);
     this.liveInfos = new ConcurrentSkipListMap<>();
     this.liveInfoLock = new ReentrantLock();
-    if (kubeResourceType == KubeResourceType.DEPLOYMENT) {
-      this.objectMetaProvider = (ObjectMetaProvider<T>) new DeploymentMetaProvider();
```

Comment:
This cast generates the unchecked warning. However since we know that DeploymentMetaProvider is of type ObjectMetaProvider<V1Deployment> it is safe to supress it. Moved the creation to the separate small body method so that warning can be supressed.